test/AutocompleteBadges.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+import { async } from 'metal';
+import AutocompleteBadges from '../src/AutocompleteBadges';
+import dom from 'metal-dom';
+
+var component;
+var input;
+
+var elements = ['Alabama', 'Arizona', 'California', 'Colorado', 'Florida', 'Indiana'];
+
+describe('AutocompleteBadges', function() {
+
+	afterEach(function() {
+		if (component) {
+			component.dispose();
+		}		
+	});
+
+	it('should check display component', function() {
+		component = new AutocompleteBadges({
+			elements: elements
+		});
+
+		assert.ok(component.visible);
+	});
+
+	it('should check elements length greater than zero with valid query', function() {
+		component = new AutocompleteBadges({
+			elements: elements
+		});
+
+		assert.ok(component.getFilteredElements_('a').length > 0);
```

Comment:
It's best to avoid calling private/protected functions in tests. The main idea for the tests is to test the public api that can be accessed by developers using the component. So, instead of calling `getFilteredElements_` to see that there are no elements, it's best to trigger the ui that shows the autocomplete list items by typing `'a'` in the textarea and check that it's not showing up (since it's empty). Same goes for the other tests that are using private/protected functions like this.